Deed helper

Phemus opened the pull request.

Commit:
rewrote create case to not set deed id

Commit:
reverted last change

Commit:
added missing variable

Commit:
removed deed id from payload

Commit:
updated to add create deed helper

Commit:
updated config file for deed api

Commit:
added error handling

endamccormack commented on app/case/deed_created.mustache:
```diff
@@ -0,0 +1,15 @@
+<main id="content" role="main">
+
+  <div class="inner-block">
```
this div and its close div are not on the same indent.

endamccormack commented on app/case/views.py:
```diff
@@ -11,6 +11,9 @@ def __init__(self, cases):
 class Start(Template):
     pageTitle = "View your case list"
 
```
have you ran flake8? I think it would complain about 1 line after class.

Phemus replied:
I have flake8 installed in my ide and it didn't pick this one up but will check it

endamccormack commented on app/service/case_api.py:
```diff
@@ -16,7 +16,7 @@ def get_cases(self):
         payload = {
             "conveyancer_id": 1,
-            "deed_id": int(deed_id)
+            "deed_id": 1
```
you removed the need for deed_id but its still in the method signature

michaeldfallen commented on app/service/case_api.py:
```diff
@@ -24,3 +24,15 @@ def create_case(self, deed_id):
+        payload = {
+            "deed_id": deed_id
+        }
+        endpoint = self.case_endpoint + "/" + case_id + "/deed"
```
Can you use string formatting over concatenation? It prevents some obscure type problems that can occur:

```
"{case}/{case_id}/deed".format(
    case=self.case_endpoint,
    case_id=case_id
)
```

michaeldfallen commented on app/case/routes.py:
```diff
@@ -29,3 +29,10 @@ def create_case():
 
         return views.Start().render()
+
+    @blueprint.route('/case/<case_id>/deed/new')
```
The create deed transaction is going to be a full protracted transaction for conveyancers so can you move this in to a new blueprint that will contain all the "create deed" stuff.

michaeldfallen commented:
When we were talking about the journey last we talked about how it's going to become a new full transaction with multiple steps. So it would be useful to follow the `click create deed` -> `Taken to create deed transaction` -> `click create deed` -> `return to caselist` flow. That way it's easy to slip steps in to the middle of that flow.

dawidskainos committed:
deed helper
 - unit tests
 - create deed page added
 - creating case without deed_id
 - create deed button exist only when case has no deed

endamccormack commented on app/deed/deed_created.mustache:
```diff
@@ -0,0 +1,15 @@
+<main id="content" role="main">
+
+  <div class="inner-block">
```
This is under indented

dawidskainos committed:
indentation

michaeldfallen commented:
:+1: Merging
